securemail/securemail.php

```diff
@@ -1,92 +1,175 @@
-	$uid = $b['uid'];
-
-    $enable_checked = get_pconfig($uid,'securemail','enable');
-    if (!$enable_checked) return;
-
-    $public_key_ascii = get_pconfig($uid,'securemail','pkey');
-
-    $gpg = new GPG();
-
-    # create an instance of a GPG public key object based on ASCII key
-    $pub_key = new GPG_Public_Key($public_key_ascii);
-
-    # using the key, encrypt your plain text using the public key
-    $txt_encrypted = $gpg->encrypt($pub_key,$b['textVersion']);
-    #$html_encrypted = $gpg->encrypt($pub_key,$b['htmlVersion']);
-
-    $b['textVersion'] = $txt_encrypted;
-    $b['htmlVersion'] = null;
-}
+<?php
+/**
+ * Name: Secure Mail
+ * Description: Send notification mail encrypted with user-defined public GPG key
+ * Version: 2.0
+ * Author: Fabio Comuni <http://kirgroup.com/profile/fabrixxm>
+ */
+
+require_once "include/Emailer.php";
+
+/* because the fraking openpgp-php is in composer, require libs in composer
+ * and then don't use autoloader to load classes... */
+$path = __DIR__."/vendor/phpseclib/phpseclib/phpseclib/";
+set_include_path(get_include_path() . PATH_SEPARATOR . $path);
+/* so, we don't use the autoloader and include what we need */
+$path = __DIR__."/vendor/singpolyma/openpgp-php/lib";
+set_include_path(get_include_path() . PATH_SEPARATOR . $path);
+
+require_once "openpgp.php";
+require_once "openpgp_crypt_symmetric.php";
+
+
+function securemail_install() {
+    register_hook('plugin_settings', 'addon/securemail/securemail.php', 'securemail_settings');
+    register_hook('plugin_settings_post', 'addon/securemail/securemail.php', 'securemail_settings_post');
+
+    register_hook('emailer_send_prepare',  'addon/securemail/securemail.php', 'securemail_emailer_send_prepare');
+
+    logger("installed securemail");
+}
+
+function securemail_uninstall() {
+    unregister_hook('plugin_settings', 'addon/securemail/securemail.php', 'securemail_settings');
+    unregister_hook('plugin_settings_post', 'addon/securemail/securemail.php', 'securemail_settings_post');
+
+    unregister_hook('emailer_send_prepare',  'addon/securemail/securemail.php', 'securemail_emailer_send_prepare');
+
+    logger("removed securemail");
+}
+
+
+function securemail_settings(&$a,&$s){
+    if(! local_user()) {
+        return;
+    }
+
+    $enable = intval(get_pconfig(local_user(),'securemail','enable'));
+    $publickey = get_pconfig(local_user(),'securemail','pkey');
+
+    $t = get_markup_template( "admin.tpl", "addon/securemail/" );
+
+    $s = replace_macros($t, array(
+        '$title' => t('"Secure Mail" Settings'),
+        '$submit' => t('Save Settings'),
+        '$test' => t('Save and send test'), //NOTE: update also in 'post'
+        '$enable' => array('securemail-enable', t('Enable Secure Mail'), $enable, ""),
+        '$publickey' => array('securemail-pkey', t('Public key'), $publickey, t("Your public PGP key, ascii armored format"), "rows='10'")
+    ));
+
+
+
+    return;
+}
+function securemail_settings_post(&$a, &$b){
```

Comment:
Standards: Please add a new line after the last function, type-hint to arguments and a leading documentation comment block.